lib/dune

```diff
@@ -0,0 +1,6 @@
+(library
+  (name core_lwt)
+  (public_name core-lwt)
```

Comment:
I'm not sure that my understanding of the `public_name` stanza is correct, but I think it should `core_lwt`, because this is the name it was known in the [oasis][1] and this is how it was referenced. 

[1]: https://github.com/BinaryAnalysisPlatform/core-lwt/pull/5/files#diff-0b6213b84518a662543be7b7516bc846L13

Reply:
It should be still `core_lwt`, because this is the name of the library as seen by ocamlfind, specified in meta, etc.